### PR TITLE
4335 error for failed save deletion

### LINKED ISSUE #4355: Failure to delete a save should give an error popup
Right now it just gets logged in `SaveHelper.DeleteSave`

### PR DESCRIPTION
**Pull request by JonFazliu**
**Brief Description of What This PR Does**

This PR adds an error message if the save fails to be deleted

**Related Issues**

close #4355

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [X] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/saving/SaveList.cs
Comment (Kastsk) [inline src/saving/SaveList.cs:72]: While it's not entirely wrong to use `ErrorDialog`, the actual established practice for displaying errors for this kinds of operation is just to use our plain `CustomConfirmationDialog` for friendlier user interface. For more info, see #3947.

I know this is counter-intuitive but take it as an example of [Maslow's hammer](https://en.wiktionary.org/wiki/if_all_you_have_is_a_hammer,_everything_looks_like_a_nail), I guess.

File: src/saving/SaveList.cs
Comment (Kastsk) [inline src/saving/SaveList.cs]: Instead, show the exception message (containing the file's full path) in the Godot error print, which can be added below this.

Also, displaying the stack trace won't do much as we as programmers already know where and how the exception occurs and the error won't change at all in all of the game's lifetime. Let alone showing it to the player.

File: src/saving/SaveList.tscn
Comment (Kastsk) [inline src/saving/SaveList.tscn:135]: The consequence of doing all of the things above is that we have a static error text, hence it's more appropriate to set the error dialog's title and message from the inspector instead of setting it programmatically. With one catch: we fill out a translation key instead of plain English text, for more info see here: [Working With Translations](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/working_with_translations.md).

**Review by Kastsk — CHANGES_REQUESTED** [review]
Welcome and thanks for opening your first PR for Thrive. I have a few comments on the changes.

File: src/saving/SaveList.cs
Comment (Kastsk) [reply src/saving/SaveList.cs]: Forgot to mention that the exception message (which in this case is a file path), shouldn't also be shown, this is related to the same choices made in #3947.

File: src/saving/SaveHelper.cs
Comment (JonFazliu) [reply src/saving/SaveHelper.cs]: That makes sense, I went ahead and made the change

File: src/saving/SaveList.cs
Comment (hhyyrylainen) [reply src/saving/SaveList.cs]: I see some problems with the translation system, so I'll just drop this link here: https://github.com/Revolutionary-Games/Thrive/blob/master/doc/working_with_translations.md

Comment (Kastsk) [reply src/saving/SaveList.cs]: Already ahead of you https://github.com/Revolutionary-Games/Thrive/pull/4431#discussion_r1271449289 🙂

File: src/saving/SaveList.tscn
Comment (JonFazliu) [reply src/saving/SaveList.tscn]: I changed the ErrorMessages to CustomConfirmationDialog, and added the title and description keys in the en.po file, though I'm a bit worried I broke something when I added them there

Comment (Kastsk) [reply src/saving/SaveList.tscn]: It looks fine from a cursory look and looks okay now overall. I'll check back tomorrow (within my timezone).

**Review by Kastsk — APPROVED** [review]
Well, this works correctly and as intended in my test, code changes are also good now so I'm approving this. Thanks again for contributing!

File: src/saving/SaveManagerGUI.cs
Comment (hhyyrylainen) [inline src/saving/SaveManagerGUI.cs:45]:
```suggestion
    public NodePath SaveDeletionFailedErrorPath = null!;
```

Ordering the words like this makes it sound natural to my ears. The current word order sounds really clunky to me. I think in the SaveList.cs there's similar kind of non-optimal word order used in the variable name.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
The approach seems fine to me but I'd like to see some slight tweaks to the specifics before merging.

**Comment by lawsonnuland** [comment]
@JonFazliu are you planning on continuing this PR?

**Comment by hhyyrylainen** [comment]
It's probably safe to assume now that this won't be continued anytime soon. It should be fine for someone else to take over this PR (note that anyone without write access to the main repo also won't be able to push to this PR branch so a new PR needs to be opened, optionally containing the commits from here as a base).

**Comment by CoreyHendrey** [comment]
Work continued over at:
https://github.com/Revolutionary-Games/Thrive/pull/4485

**Pull request closed without merging**
